libavcodec/vaapi_encode.c

```diff
@@ -2052,6 +2060,7 @@ static av_cold int vaapi_encode_init_rate_control(AVCodecContext *avctx)
 #if VA_CHECK_VERSION(1, 1, 0)
             .ICQ_quality_factor = av_clip(rc_quality, 1, 51),
             .max_qp             = (avctx->qmax > 0 ? avctx->qmax : 0),
+            .rc_flags.bits.mb_rate_control = ctx->mbbrc ? 1 : 2,
```

Comment:
Why do we not use the default value (0) ?

Reply:
MBBRC brings higher quality to objects in picture, but drops frame level PSNR. So I'd prefer to enable/disable it in ffmpeg if user really understand its pros and cons instread of decided automatically by driver. And for VPL, it also disabled by default for some hw/codecs.

Reply:
> And for VPL, it also disabled by default for some hw/codecs.
So it is enabled for some hw/codecs, right ?

Reply:
> > And for VPL, it also disabled by default for some hw/codecs.
> 
> So it is enabled for some hw/codecs, right ?

Only auto(unknow) and disable by default from ffmpeg vpl log.

Reply:
Does setting mb_rate_control to 1 or 2 make sense while the underlying driver doesn't support VA_RC_MB ?

Reply:
There is code to check supported VA_RC_MB, if not, mb_rate_control will be set to 1 only.